DigitalLearningSolutions.Data.Tests/Helpers/LearningHubSsoSecurityHelperTests.cs

```diff
@@ -0,0 +1,160 @@
+﻿namespace DigitalLearningSolutions.Data.Tests.Helpers
+{
+    using System;
+    using DigitalLearningSolutions.Data.Helpers;
+    using DigitalLearningSolutions.Data.Services;
+    using FluentAssertions;
+    using NUnit.Framework;
+
+    public class LearningHubSsoSecurityHelperTests
+    {
+        private const string Secret = "where the wild rose blooms";
+
+        [Test]
+        public void GenerateHash_is_consistent()
+        {
+            // given
+            var now = DateTime.UtcNow;
+            var stateString = "stateString";
+            var clockService = new BinaryClockService(now, now);
+            var helper = new LearningHubSsoSecurityHelper(clockService);
+
+            // when
+            var hash1 = helper.GenerateHash(stateString, Secret);
+            var hash2 = helper.GenerateHash(stateString, Secret);
+
+            // then
+            hash1.Should().BeEquivalentTo(hash2);
+        }
+
+        [Test]
+        public void GenerateHash_varies_hash_with_clock_time()
+        {
+            // given
+            var now = DateTime.UtcNow;
+            var stateString = "stateString";
+            var clockService = new BinaryClockService(now, now.AddSeconds(1));
+            var helper = new LearningHubSsoSecurityHelper(clockService);
+
+            // when
+            var hash1 = helper.GenerateHash(stateString, Secret);
+            var hash2 = helper.GenerateHash(stateString, Secret);
+
+            // then
+            hash1.Should().NotBeEquivalentTo(hash2);

```

Comment:
I think case matters here and so `Should().NotBe<string>)` would be more appropriate.
`NotBeEquivalentTo` is case insensitive.